test/dune

```diff
@@ -163,3 +163,21 @@
  (name   runtest)
  (deps   (:x section.out) (:y section.out.expected))
  (action (diff? %{x} %{y})))
+
+; Test failures
+
+(rule
+ (targets failures.outputs)
+ (deps    (package mdx) failures/empty.ml)
+ (action
+  (with-outputs-to %{targets}
+   (progn
+    (system "%{bin:ocaml-mdx} test \"%{dep:failures/direction_not_set.md}\" || echo")
```

Comment:
I'm adding quotes to avoid problems with spaces. But some characters will still be interpreted by bash (`$`, backquote, `\`, `"`) and cmd (`%`, `!`).
I don't known any syntax that would fix that and be parsed by both bash and cmd.

Reply:
if that's just for the tests, maybe it's not necessary to add `"` as well.

Reply:
Yes. Looks like `dune` only use relative paths so this should not be a problem.
Reverted.
Still, the ideal would be to have an equivalent of `(run ..)` that don't check exit code.

Reply:
Or that checks that it fails!

Reply:
If #178 is merged, we could do something similar to the solution in Alcotest for this: https://github.com/mirage/alcotest/blob/master/test/e2e/expect_failure.ml. Agreed that it would be nice for the `dune` `run` DSL to support this; I've been meaning to make an issue about this for a while.

Reply:
Yes that would be great. I'll PR your PR.